exercises/sort_by_pattern_test.rb

```diff
@@ -46,23 +47,44 @@ def test_sort_by_distance
   end
 
   def test_sort_by_length
-    skip
     words = ["heteromorph", "ancyloceratina", "bioengineering", "mathematical", "bug"]
-    # Your code goes here
+    transformed = []
+    words.each do |word|
+      transformed << [word.length, word]
+    end
+    transformed.sort!
+    sorted = []
+    transformed.each do |sort_key, word|
+      sorted << word
+    end
     assert_equal ["bug", "heteromorph", "mathematical", "ancyloceratina", "bioengineering"], sorted
   end
 
   def test_sort_by_proximity_to_ten
-    skip
     prices = [3.02, 9.91, 17.9, 10.01, 11.0]
-    # Your code goes here
+    transformed = []
+    prices.each do |price|
+      transformed << [(10.0 - price).abs, price]
+    end
+    transformed.sort!.reverse
+    sorted = []
+    transformed.each do |sort_key, price|
+      sorted << price
+    end
     assert_equal [10.01, 9.91, 11.0, 3.02, 17.9], sorted
   end
 
   def test_sort_by_number_of_cents
-    skip
     prices = [3.02, 9.91, 7.9, 10.01, 11.0]
-    # Your code goes here
+    transformed = []
+    prices.each do |price|
+      transformed << [price.to_s.split(".").pop, price]
+    end
+    transformed.sort!
+    sorted = []
+    transformed.each do |sort_key, price|
```

Comment:
Unused block argument - `sort_key`. If it's necessary, use `_` or `_sort_key` as an argument name to indicate that it won't be used.